x/accounts/vesting/vesting.go

```diff
@@ -0,0 +1,490 @@
+			if err != nil {
+				return nil, err
+			}
+
+			err = bva.TrackDelegation(
+				ctx,
+				sdk.Coins{*resp.Balance},
+				vestingCoins,
+				sdk.Coins{msgDelegate.Amount},
+			)
+			if err != nil {
+				return nil, err
+			}
+		case "/cosmos.staking.v1beta1.MsgUndelegate":
+			msgUndelegate, err := accountstd.UnpackAny[stakingtypes.MsgUndelegate](m)
+			if err != nil {
+				return nil, fmt.Errorf("invalid proto msg for type: %s", typeUrl)
+			}
+
+			err = bva.TrackUndelegation(ctx, sdk.Coins{msgUndelegate.Amount})
+			if err != nil {
+				return nil, err
+			}
+		case "/cosmos.bank.v1beta1.MsgSend", "/cosmos.bank.v1beta1.MsgMultiSend":
+			var sender string
+			var amount sdk.Coins
+			if typeUrl == "/cosmos.bank.v1beta1.MsgSend" {
+				msgSend, err := accountstd.UnpackAny[banktypes.MsgSend](m)
+				if err != nil {
+					return nil, fmt.Errorf("invalid proto msg for type: %s", typeUrl)
+				}
+				sender = msgSend.FromAddress
+				amount = msgSend.Amount
+			} else {
+				msgMultiSend, err := accountstd.UnpackAny[banktypes.MsgMultiSend](m)
+				if err != nil {
+					return nil, fmt.Errorf("invalid proto msg for type: %s", typeUrl)
+				}
+				sender = msgMultiSend.Inputs[0].Address
+				amount = msgMultiSend.Inputs[0].Coins
+			}
+
+			vestingCoins, err := getVestingFunc(ctx, hs.Time)
+			if err != nil {
+				return nil, err
+			}
+
+			// Get locked token
+			lockedCoins := bva.LockedCoinsFromVesting(ctx, vestingCoins)
+
+			// Check if any sent tokens is exceeds vesting account balances
+			for _, coin := range amount {
+				// Query account balance for the sent denom
+				balanceQueryReq := banktypes.NewQueryBalanceRequest(sdk.AccAddress(sender), coin.Denom)
+				resp, err := accountstd.QueryModule[banktypes.QueryBalanceResponse](ctx, balanceQueryReq)
+				if err != nil {
+					return nil, err
+				}
+				balance := resp.Balance
+				locked := sdk.NewCoin(coin.Denom, lockedCoins.AmountOf(coin.Denom))
+
+				spendable, hasNeg := sdk.Coins{*balance}.SafeSub(locked)
+				if hasNeg {
+					return nil, errorsmod.Wrapf(sdkerrors.ErrInsufficientFunds,
+						"locked amount exceeds account balance funds: %s > %s", locked, balance)
+				}
+
+				if _, hasNeg := spendable.SafeSub(coin); hasNeg {
+					if len(spendable) == 0 {
+						spendable = sdk.Coins{sdk.NewCoin(coin.Denom, math.ZeroInt())}
+					}
+					return nil, errorsmod.Wrapf(
+						sdkerrors.ErrInsufficientFunds,
+						"spendable balance %s is smaller than %s",
+						spendable, coin,
+					)
+				}
+			}
+		}
+	}
+
+	// execute messages
+	responses, err := accountstd.ExecModuleAnys(ctx, msg.ExecutionMessages)
+	if err != nil {
+		return nil, err
+	}
+
+	return &account_abstractionv1.MsgExecuteResponse{ExecutionMessagesResponse: responses}, nil
+}
+
+// Check the sender of the bundle is the owner of the vesting account
+func (bva BaseVesting) Authenticate(ctx context.Context, msg *account_abstractionv1.MsgAuthenticate) (*account_abstractionv1.MsgAuthenticateResponse, error) {
```

Comment:
this is not correctly implemented, this should do the whole auth signing flow (wrapping a base account); i would suggest for this to be removed